src/main/java/com/epam/test_generator/web_security/WebSecurityConfig.java

```diff
@@ -0,0 +1,79 @@
+package com.epam.test_generator.web_security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+
+@EnableWebSecurity
+@Configuration
+
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
```

Comment:
Please provide full documentation on application security. Documentation should include every class that participates in authorization & authentication flow, how they interact with each other, how they interact with controllers, and most importantly order of jwt filters/services execution during processing client request. Also document all of added classes. In documentation focus on WHY this entity was added, rather then on WHAT it does.

Reply:
+